binrw_derive/src/binrw/parser/mod.rs

```diff
@@ -40,12 +42,17 @@ trait FromAttrs<Attr: syn::parse::Parse> {
                     is_binread_attr(attr)
                 }
             })
-            .flat_map(
-                |attr| match syn::parse2::<MetaAttrList<Attr>>(attr.tokens.clone()) {
+            .flat_map(|attr| {
+                let tokens = match &attr.meta {
+                    syn::Meta::Path(_) => TokenStream::new(),
+                    syn::Meta::List(ml) => ml.tokens.clone(),
+                    syn::Meta::NameValue(nv) => nv.to_token_stream(),
```

Comment:
This looks great! Really the only unaddressed question I have is regarding these new match arms. I believe that it had been the case that `MetaAttrList` would have failed a parse in `parenthesized!` before, but now that part of it has been uplifted into `syn::Meta`. I notice there is now a `syn::Meta::require_list` function. It seems to me like that should be used here instead?